src/pairwise_dist/_pairwise_dist.pyx

```diff
@@ -0,0 +1,46 @@
+# cython: boundscheck=False
+
+cimport numpy as np
+import numpy as np
+from libc.math cimport fabs
+from cython.parallel cimport prange
+from cython cimport floating, integral
+cimport cython
+
+cdef void _compute_dist(
+    floating[::1] X_a_row,
+    floating[::1] X_b_row,
+    integral n_features,
+    floating *dist
+) nogil:
+    cdef integral i
+    for i in range(n_features):
+        dist[0] += fabs(X_a_row[i] - X_b_row[i])
+
+cdef void _pairwise_dist(
+    floating[:, ::1] X_a, # IN
+    floating[:, ::1] X_b, # IN
+    floating[:, ::1] distances, # OUT
+    integral dummy
```

Comment:
As discussed, I don't know what's the best here to make Cython able to specialize functions for integral.
Using `int` is or even `cnp.intp_t` types should be fine as we solely used them for indexing.